system/ee/ExpressionEngine/Addons/channel/libraries/channel_form/Channel_form_lib.php

```diff
@@ -2705,7 +2705,12 @@ public function initialize($reinitialize = false)
         );
 
         $this->custom_fields = array();
-        $this->custom_option_fields = array();
+
+        if ($custom_option_fields = ee('Model')->get('Config')->filter('key', 'custom_option_fields')->filter('site_id', $this->site_id)->first()) {
+            $this->custom_option_fields = json_decode($custom_option_fields->value);
```

Comment:
Will this work correctly if `$custom_option_fields->value` is empty, or not defined property?